implementation/src/main/java/io/smallrye/jwt/auth/principal/KeyLocationResolver.java

```diff
@@ -52,50 +53,56 @@
  * This implements the MP-JWT 1.1 mp.jwt.verify.publickey.location config property resolution logic
  */
 public class KeyLocationResolver implements VerificationKeyResolver {
-    private static final Logger log = Logger.getLogger(KeyLocationResolver.class);
+    private static final Logger LOGGER = Logger.getLogger(KeyLocationResolver.class);
 
     // The 'content' and 'httpsJwks' fields are used to keep the key content and mutually exclusive.
     // 'content' represents the key(s) loaded from all resources but the HTTPS URL based JWK set.
-    private String content;
+    private final String content;
     // 'httpsJwks' represents the JWK set loaded from the HTTPS URL.
-    private HttpsJwks httpsJwks;
+    private final HttpsJwks httpsJwks;
+    private final JWTAuthContextInfo authContextInfo;
 
     // If the 'smallrye.jwt.token.kid' is set then the verification key will be calculated
     // only once and used for all the token verification requests
     private volatile PublicKey verificationKey;
 
-    private JWTAuthContextInfo authContextInfo;
-
     public KeyLocationResolver(JWTAuthContextInfo authContextInfo) throws UnresolvableKeyException {
         this.authContextInfo = authContextInfo;
         try {
-            loadContents();
+            AbstractMap.SimpleImmutableEntry<HttpsJwks, String> contents = loadContents();
```

Comment:
I'm not sure we need this kind of clean up, The key and value type are never used at the same time so it looks a bit unnatural to me, even though, anti-pattern wise, this style is technically cleaner.

Reply:
Would it be cleaner to have `loadContents` continue to set the instance variables? I agree it looks unnatural to use `Map.Entry` as a way to return two values that are not actually a key/value pair.

Reply:
@MikeEdgar agreed